dist/tools/genconfigheader/genconfigheader.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+#
+# Copyright (C) 2016 Eistec AB
+#
+# This file is subject to the terms and conditions of the GNU Lesser General
+# Public License v2.1. See the file LICENSE in the top level directory for more
+# details.
+#
+
+if [ $# -lt 1 ]; then
+  echo "Usage: $0 <output.h> [CFLAGS]..."
+  echo "Extract all macros from CFLAGS and generate a header file"
+  exit 1
+fi
+OUTPUTFILE="$1"
+shift
+
+MD5SUM=md5sum
+UNAME=$(uname -s)
+if [ "$UNAME" = "Darwin" ]; then
+  MD5SUM=md5 -r
+fi
+
+# atomically update the file
+TMPFILE=
+trap '[ -n "${TMPFILE}" ] && rm -f "${TMPFILE}"' EXIT
+# Create temporary output file
+TMPFILE=$(mktemp ${OUTPUTFILE}.XXXXXX)
+
+if [ -z "${TMPFILE}" ]; then
+  echo "Error creating temporary file, aborting"
+  exit 1
+fi
+
+# exit on any errors below this line
+set -e
+
+echo "/* Automatically generated header, changes will be overwritten */" > "${TMPFILE}"
```

Comment:
probably we should give a word not to edit this file directly, since it has no effect:
```
echo "/* DO NOT edit this file, your changes will be overwritten and won't take any effect! */" > "${TMPFILE}"
```